src/back/spv/writer.rs

```diff
@@ -898,53 +897,129 @@ impl Writer {
                             )
                             .unwrap();
 
-                        let left_ty_inner = &ir_module.types[left_ty.unwrap()].inner;
-                        let right_ty_inner = &ir_module.types[right_ty.unwrap()].inner;
+                        let left_ty = left_ty.unwrap();
+                        let right_ty = right_ty.unwrap();
 
-                        let (result_type_id, vector_id, scalar_id) =
-                            match (left_ty_inner, right_ty_inner) {
-                                (
-                                    crate::TypeInner::Vector { .. },
-                                    crate::TypeInner::Scalar { .. },
-                                ) => (
-                                    self.get_type_id(
-                                        &ir_module.types,
-                                        LookupType::Handle(left_ty.unwrap()),
-                                    ),
+                        let left_ty_inner = &ir_module.types[left_ty].inner;
+                        let right_ty_inner = &ir_module.types[right_ty].inner;
+
+                        let left_result_type_id =
+                            self.get_type_id(&ir_module.types, LookupType::Handle(left_ty));
+
+                        let right_result_type_id =
+                            self.get_type_id(&ir_module.types, LookupType::Handle(right_ty));
+
+                        let left_id = match *left_expression {
+                            crate::Expression::LocalVariable(_)
+                            | crate::Expression::GlobalVariable(_) => {
```

Comment:
Little bit dirty this way, but these matches were already there in other expressions, but it will suffice for now.

Reply:
This is something we'll need to do for *any* variable used in any expression. So a helper method is needed, and we need to call this from more places. Do you want to leave this for a follow-up?

Reply:
Yes please, I will take a note and store it as task in my todo.